.circleci/config.yml

```diff
@@ -88,24 +63,28 @@ jobs:
           name: Build and deploy to AppCenter
           command: bundle exec fastlane appcenter
 
+orbs:
+  path-filtering: circleci/path-filtering@0.1.1
+
 workflows:
-  version: 2
 
   build-testflight-deploy:
     jobs:
-      - build-testflight-deploy:
+      - path-filtering/filter:
           filters:
-            tags:
-              only: /^v.*/
             branches:
-              ignore: /.*/
+              only:
+                - /^.*main.*/
+          name: Determine if a release deploy is needed
+          config-path: .circleci/deploy-release.yml
+          mapping: |
+            fastlane/VERSION deploy-release true
```

Comment:
When making up the new process, I thought about that case too. While it seems to be a very valid point, it challenges the process of merging into `main` itself.
As per our process, merging in `main` before QAing put some tradeoffs on the table.
Our release process should wait for a "stable/releasable" `main` to spin up a release.
Before this new method, it would have been simpler to trigger a CI release process by tagging a version. However, the robustness of the process itself was validated because of factors like team size shared knowledge of one-two individual. While it can be applied now in some ways as well, the intention of this process is to streamline it to potentially everyone capable of updating a file with a new version number and opening a PR, straight from GitHub website.

Reply:
Makes sense, either way the expected flow is to only merge to main stable branches and to wait for tickets raised during QA to be fixed before deploying anyway 👍 I'm fine with setting this as the process and iterating on it if we need any different flow in the future 🙂

Reply:
Exactly 👍 
merging straight into `main` embraces the extreme developer approach - I'm also happy to leave it as is and then review it if needed. In case of awaiting before merge we'll need app center builds to be triggered based on feature branches